iquip/apps/dataviewer.py

```diff
@@ -238,3 +240,169 @@ def __init__(self, parent: Optional[QWidget] = None):
         layout.addLayout(buttonGroupLayout)
         layout.addWidget(self.stack)
         self.buttonGroup.idClicked.connect(self.stack.setCurrentIndex)
+
+
+class DataPointWidget(QWidget):
+    """Widget for configuring each data point.
+    
+    Attributes:
+        seriesLabel: Label showing the name of the current data series (dataset).
+        numberOfSamplesBox: Spin box showing the total number of samples.
+        thresholdBox: Spin box for setting the threshold for state discrimination.
+        buttonGroup: Data type selection radio button group.
+        valueBoxes: Dict of spin boxes for each data type.
+
+    Signals:
+        dataTypeChanged(dataType): Current data type is changed to dataType.
+        thresholdChanged(threshold): Current threshold value is changed to threshold.
+    """
+
+    class DataType(enum.IntEnum):
+        """Type of each data point.
+        
+        Each item is used as its index in the button group.
+        """
+        TOTAL = 0
+        AVERAGE = 1
+        P1 = 2
+
+    dataTypeChanged = pyqtSignal(DataType)
+    thresholdChanged = pyqtSignal(int)
+
+    def __init__(self, parent: Optional[QWidget] = None):
+        """Extended."""
+        super().__init__(parent=parent)
+        layout = QGridLayout(self)
+        # first column (general information)
+        self.seriesLabel = QLabel("", self)
```

Comment:
If you want to set the initial text to an empty string, just `QLabel(self)` looks better!